app/test/unit/promise-with-timeout-test.ts

```diff
@@ -0,0 +1,62 @@
+import { expect } from 'chai'
+
+import { promiseWithMinimumTimeout } from '../../src/lib/promise'
+
+jest.useFakeTimers()
+
+describe('promiseWithMinimumTimeout', () => {
+  it('handles promise finishing before timeout', async () => {
+    const fastPromise = new Promise<number>((resolve, reject) => {
+      window.setTimeout(() => {
+        resolve(42)
+      }, 100)
+    })
+
+    const promise = promiseWithMinimumTimeout(() => fastPromise, 500)
+
+    // promise completes
+    jest.advanceTimersByTime(250)
+    // timeout completes
+    jest.advanceTimersByTime(250)
+
+    const result = await promise
+
+    expect(result).equals(42)
+  })
+
+  it('handles promise and timeout finishing together', async () => {
+    const mediumPromise = new Promise<number>((resolve, reject) => {
+      window.setTimeout(() => {
+        resolve(42)
+      }, 500)
+    })
+
+    const promise = promiseWithMinimumTimeout(() => mediumPromise, 500)
+
+    // both complete
+    jest.advanceTimersByTime(500)
+
+    const result = await promise
+
+    expect(result).equals(42)
+  })
+
+  it('handles promise finishing after timeout', async () => {
+    const slowPromise = new Promise<number>((resolve, reject) => {
+      window.setTimeout(() => {
+        resolve(42)
+      }, 1000)
+    })
+
+    const promise = promiseWithMinimumTimeout(() => slowPromise, 500)
+
+    // timeout completes
+    jest.advanceTimersByTime(500)
```

Comment:
should we check that `promise` isn't resolved right after this?

Reply:
I've pushed 6a2b963d7 which let's us check the state of the promise we control